src/truncateString.js

```diff
@@ -66,11 +66,13 @@ class TruncateString extends PureComponent {
 
   resetTruncate = debounce(50, () => {
     // this renders the original string so we can measure it
-    this.setState({truncating: true}, () => {
-      // now we render again with the truncated string
-      const truncatedString = this.getTruncateString(this.props.text)
-      this.setState({truncatedString, truncating: false})
-    })
+    if (this._isMounted) {
```

Comment:
One other approach here, might be to change `debounce` to a `this.timeoutId = setTimeout(...)`. When the component unmounts you can just `clearTimeout(this.timeoutId)` so this function never fires.